### ISSUE TITLE
Compatibility with New Teams??

### ISSUE THREAD

**Issue by konax3** [Pat992/teams_multi_instances#2]
Do you planned to do a new version compatible with the New Teams? 
I know the new temas have a bult-in option to manage multiple accounts by i don't like so much, i don't like to switch for with account to other.

Thanks in advance.

**Comment by Pat992** [comment]
Hi
Yes I have planned to update the app for the new version of Teams.
Are using the Teams-App from the Microsoft store?

Best regards

**Comment by konax3** [comment]
No.
I use the desktop app and a few months ago, a new button appears in the top left corner to try the new version. I tryed it but I came back to the current one because I prefer two separate instances.

![image](https://github.com/Pat992/teams_multi_instances/assets/63001964/fb0499d0-42af-4dc7-a870-103032406144)

**Comment by Pat992** [comment]
I see, so if I understand you correctly, when you try to use my app with the new teams enabled, it will not open a new instance?

Can you check if you have a Teams-Folder in "C:\Program Files\WindowsApps\", the folder should be called something like "MicrosoftTeams_23231.411.2342.9597_x64__8wekyb3d8bbwe".

There are two ways to check for this folder, the first one is to run the following command with Powershell as an Admin:
`Get-ChildItem -Path 'C:\Program Files\WindowsApps\MicrosoftTeams*\*'`
You should then see a list of files in your Powershell-Terminal.

Or if you have Teams open, you can open the directory in the task-manager:
![grafik](https://github.com/Pat992/teams_multi_instances/assets/50899306/b79e4062-fead-4ead-8384-9bee997f8822)

Best regards

**Comment by konax3** [comment]
> I see, so if I understand you correctly, when you try to use my app with the new teams enabled, it will not open a new instance?
> 
> Can you check if you have a Teams-Folder in "C:\Program Files\WindowsApps", the folder should be called something like "MicrosoftTeams_23231.411.2342.9597_x64__8wekyb3d8bbwe".
> 
> There are two ways to check for this folder, the first one is to run the following command with Powershell as an Admin: `Get-ChildItem -Path 'C:\Program Files\WindowsApps\MicrosoftTeams*\*'` You should then see a list of files in your Powershell-Terminal.
> 
> Or if you have Teams open, you can open the directory in the task-manager: ![grafik](https://user-images.githubusercontent.com/50899306/268069679-b79e4062-fead-4ead-8384-9bee997f8822.png)
> 
> Best regards

Yes! that's it. With the new teams button enabled your arpp doesn't opens a new instance.

Yes, you are rigth, with the new version enabled i can see this folder:

C:\Program Files\WindowsApps\MSTeams_23247.1108.2370.286_x64__8wekyb3d8bbwe

It seems is the version from windows store, right?

**Comment by Pat992** [comment]
Ok, thank you for the update.

That is good, I think the new Version of Teams is a UWP-App, while the previous one was an Electron-App.
So all the data is stored in this new folder "MSTeams_23247.1108.2370.286_x64__8wekyb3d8bbwe".

Now I have just two last questions:
1. If you deactivate "Try the new Teams", is my app working again?
2. Do you have the Folder "C:\Users\<YOUR_USERNAME>\AppData\Local\Microsoft\Teams" and inside there the "Update.exe"?

Best regards

**Comment by konax3** [comment]
> Ok, thank you for the update.
> 
> That is good, I think the new Version of Teams is a UWP-App, while the previous one was an Electron-App. So all the data is stored in this new folder "MSTeams_23247.1108.2370.286_x64__8wekyb3d8bbwe".
> 
> Now I have just two last questions:
> 
> 1. If you deactivate "Try the new Teams", is my app working again?
> 2. Do you have the Folder "C:\Users<YOUR_USERNAME>\AppData\Local\Microsoft\Teams" and inside there the "Update.exe"?
> 
> Best regards

Yes, if deactivate works again perfectly. Right now i'm using it.

And yes, there is an Update.exe in this path

**Comment by Pat992** [comment]
Thank you for your response.

I will update the app as soon as possible to make it work with the older and newer version of Teams.
As soon as I have tested the changes successfully, I let you know.

Best regards

**Comment by marcin-github** [comment]
Hi, any chance for progress?

**Comment by Pat992** [comment]
Hi there,

it might be possible to start multiple instances with the `New-PSSession`,
for that the `lib/bloc/process/process_bloc.dart` would need to be updated.

Sadly I don't use Microsoft Teams and Windows anymore, if anyone is interested in fixing this, feel free to create a pull request.
